src/functions/ensureEnv.ts

```diff
@@ -124,27 +139,33 @@ export async function ensure(options: any): Promise<void> {
   );
 
   const userEnvs = await getUserEnvs(projectId);
-  if (userEnvs.length > 0) {
+  if (Object.keys(userEnvs).length > 0) {
     let msg =
       "If you opt in, the following environment variables will be deleted on next deploy:\n";
 
-    const allKvs: Record<string, string> = {};
-    for (const { fnLabel, envs } of userEnvs) {
-      msg += `\t${fnLabel}: `;
-
-      const kvs: string[] = [];
-      for (const [k, v] of Object.entries(envs)) {
-        kvs.push(clc.bold(`${k}=${v}`));
-        allKvs[k] = v;
-      }
-      msg += `${kvs.join(", ")}\n`;
-    }
-    msg += "\nTo preserve these environment variable, run the following command after opt-in:\n";
-    msg += clc.bold(
-      `\tfirebase functions:env:add ${Object.entries(allKvs)
-        .map(([k, v]) => `${k}=${v}`)
-        .join(" ")}\n`
-    );
+    // Transform userEnvs into a string of form:
+    //   helloWorld(us-central1): KEY1=VAL1, KEY2=VAL2
+    //   hellWorld(us-east1): KEY3=VAL3
+    msg += Object.entries(userEnvs)
+      .map(([label, envs]) => {
+        const envList = Object.entries(envs)
+          .map(([k, v]) => clc.bold(`${k}=${v}`))
+          .join(", ");
+        return `\t${label}: ${envList}`;
+      })
+      .join("\n");
+
+    // Transform userEnvs into a string of form:
```

Comment:
We're still not handling the case where two functions have different values for the same env. What do we even want to do here? Error out until they fix it? Just warn them in big bold letters?

Reply:
Added a little warning!